later/lib/database/datasources/records_data_source.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:later/database/models/record_model.dart';
+
+abstract class RecordsDataSource {
+  const RecordsDataSource();
+
+  Stream<List<RecordDbModel>> watchAll();
+  Future<void> create(
+      {required String? url,
+      required String title,
+      String? description,
+      DateTime? lastEditedAt});
```

Comment:
`lastEditedAt` shouldn't be here. It must only be set by the local data source itself (on creation / editing)